browser/ui/webui/settings/brave_default_extensions_handler.h

```diff
@@ -23,15 +25,23 @@ class BraveDefaultExtensionsHandler : public settings::SettingsPageUIHandler {
  private:
   // SettingsPageUIHandler overrides:
   void RegisterMessages() override;
-  void OnJavascriptAllowed() override;
-  void OnJavascriptDisallowed() override;
+  void OnJavascriptAllowed() override {}
+  void OnJavascriptDisallowed() override {}
 
   void GetRestartNeeded(const base::ListValue* args);
   void SetWebTorrentEnabled(const base::ListValue* args);
   void SetHangoutsEnabled(const base::ListValue* args);
   void SetIPFSCompanionEnabled(const base::ListValue* args);
   void SetMediaRouterEnabled(const base::ListValue* args);
   void SetBraveWalletEnabled(const base::ListValue* args);
+#if BUILDFLAG(ENABLE_TOR)
+  void SetTorDisabled(const base::ListValue* args);
+  void GetTorDisabled(const base::ListValue* args);
+  void OnTorDisabledChanged();
+#endif
+  void GetEnableTorOption(const base::ListValue* args);
```

Comment:
What is this?

Reply:
This is used to determine whether `Disable Tor` option should be enabled or not.
If tor is managed by group policy, this setting option should be in disabled state.

Reply:
If the tor feature is not included we shouldn't be showing any UI for it at all and we can detect that with the buildflag so we don't need a method for it. For group policy is should be `IsTorManaged`

Reply:
Yes, I agree with you. UI should not be visible if tor is not built.
To do this, I tried to hide settings UI by using template with `dom-if`.
However, toggle button can't get proper runtime value change if it is inside template.
I debugged this with @bsclifton but can't find solution at that time.
Maybe we need to dig more about polymer.
So, I left like this because we would not build brave w/o tor.
How about this as a f/u?

Reply:
I'm trying to find how to hide webui with buildflags.

Reply:
Moved this into tor buildflags with renamed one.
and tor settings will be hidden by adding `enable_tor` defines to grit defines.
Much more clear 👍